packages/shared_preferences/shared_preferences/tool/pre_publish.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+// ignore: depend_on_referenced_packages
```

Comment:
@stuartmorgan do you think this ignore is fine here? Didn't want to add `path` as a dependency in `shared_preferences`, although it is already a transitive dependency.

Reply:
it turns out the linter won't allow me to do that. I'll add it as a dev_dependency instead.

<img width="1261" alt="image" src="https://github.com/user-attachments/assets/5d125231-7eab-45bb-9409-f3be4f5ab32d">